src/xb-node-query.c

```diff
@@ -58,14 +59,45 @@ xb_node_query (XbNode *self, const gchar *xpath, guint limit, GError **error)
  * Returns: (transfer container) (element-type XbNode): results, or %NULL if unfound
  *
  * Since: 0.1.4
+ * Deprecated: 0.3.0: Use xb_node_query_with_bindings() instead.
```

Comment:
I don't think we want to deprecate this; I think it's easier to use than xb_node_query_with_bindings(). Basically, in the case of nothing being bound it's harmess and easier to understand.

Reply:
My thinking about deprecating as much as possible is that there are currently 13 public `*_query*` functions in libxmlb, and (imo) it’s already hard enough to work out which one you should be using. I was hoping to consolidate that a bit or, at least, not make it worse.

> Basically, in the case of nothing being bound it's harmess and easier to understand.

That is, however, true.

Reply:
Ya, it's a bit chaotic. I guess it's hard because most of the time it's "I want the cdata from this one node" rather than create query, bind values, execute query, get results etc. I think if anything needs to hit the cutting room floor, I'd rather it was the silo query functions, and we can just use the node-query in most cases and XbQuery in anything more complex.